packages/frontend/app/components/learner-group/calendar.gjs

```diff
@@ -61,22 +62,24 @@ export default class LearnerGroupCalendarComponent extends Component {
     return await map(flat, async (offering) => {
       const session = await offering.session;
       const course = await session.course;
-      return {
-        startDate: offering.startDate.toISOString(),
-        endDate: offering.endDate.toISOString(),
-        calendarStartDate: offering.startDate.toISOString(),
-        calendarEndDate: offering.endDate.toISOString(),
-        courseTitle: course.title,
-        name: session.title,
-        offering: offering.id,
-        location: offering.location,
-        color: '#84c444',
-        prerequisites: [],
-        postrequisites: [],
-        isScheduled: session.isScheduled || course.isScheduled,
-        isPublished: session.isPublished && course.isPublished,
-        isBlanked: false,
-      };
+      const school = await course.school;
+      return this.schoolEvents.createEventFromData(
```

Comment:
using the same `createEventFromData()` method across the board for creating event object from data is probably the most immediately beneficial change in this PR.
if anything here should land then it's this (alongside the same change in the offering calendar component).